### PR TITLE
haproxy: active-active mode

### PR DESCRIPTION
**Pull request by ilausuch**
Deploys the haproxy in active-active mode. All nodes in cluster will have the haproxy.

Allows the migration from previous mode active-pasive

### CODE REVIEW

**Comment by ilausuch** [comment]
[Make haproxy active/active](https://trello.com/c/7oHIy01J/271-make-haproxy-active-active)

File: chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb [hunk 1]
```diff
@@ -56,17 +74,13 @@
...
-    cib_objects lazy { transaction_objects }
-    # note that this will also automatically start the resources
-    action :commit_new
+  # Compatibility with existing deployment: we need to drop the group to create
+  # the clone
+  group_name = "g-#{service_name}"
+  pacemaker_group group_name do
+    action [:stop, :delete]
     only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
+    only_if "crm configure show group_name"
```
Comment (AbelNavarro) [inline chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb]: `#{group_name}`

Comment (ilausuch) [reply chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb]: Done

**Comment by AbelNavarro** [comment]
The tests show that when applying this patch, the connection to the API socket is down for about 4 seconds. The API socket is hold by haproxy, not by the actual API service.

Connections to the API service are down for 20 seconds when applying this patch, so it is a major change. Trials to reduce this time by including the deletion of the haproxy group with the haproxy clone in the same transaction have been unsuccessful, but anyway the improvement would have been in the best case minor.

File: chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb [hunk 2]
```diff
@@ -43,6 +43,24 @@
 end
 
 if node[:pacemaker][:haproxy][:clusters].key?(cluster_name) && node[:pacemaker][:haproxy][:clusters][cluster_name][:enabled]
+  directory "create /etc/sysctl.d for nonlocal_bind" do
```
Comment (aspiers) [inline chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb]: This shouldn't be necessary; it's provided by both `systemd` and `filesystem` rpms.

**Comment by vuntz** [comment]
Can we get the commit message fixed? :-)

File: chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb (hunk 2 above)
Comment (ilausuch) [reply chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb]: Done

**Comment by rhafer** [comment]
> Afaik, the ha_galera script completes before this line and there are sync marks assuring mysql should be running at this point, so I'm puzzled. Feedback is welcome.

@AbelNavarro  Have you looked into /var/log/messages if the galera resource agent is logging something helpful (it should if the fail count of the galera resource is > 0). Also /var/log/mysql/* might contain some helpful information. Do you still have a system where this is reproducable? We could do a debugging session.

**Comment by Itxaka** [comment]
@aspiers re-review pls?

**Comment by vuntz** [comment]
(trusting the reviews, merging to unblock)

**Pull request merged**